### ISSUE TITLE
update to support protobuf editions

### ISSUE THREAD

**Issue by zwing99** [nipunn1313/mypy-protobuf#677]
```protobuf
edition = "2023";

message TestContract {
    string id = 1;
    string name_of_person = 2;
    int32 age = 3;
}
```

using the contract above i get this warning:

`test_contracts.proto: is an editions file, but code generator protoc-gen-mypy hasn't been updated to support editions yet.  Please ask the owner of this code generator to add support or switch back to proto2/proto3.`

Editions seems like the go forward for protobuf and mypy is certainly the go forward for python. Any chance this tooling can be updated to support it?

**Comment by wbhob** [comment]
+1 on this. Editions are the future of Protobuf, will be very helpful to have this for strong types.

**Comment by nipunn1313** [comment]
yep seems reasonable! I've definitely been a bit behind in doing updates.

Let me catch some time in between things at my startup. Bear with me.

**Comment by wbhob** [comment]
Thank you — as a note, I was able to get up and running with the official protocolbuffers/pyi package. They may have a reference implementation

**Comment by HenriChabert** [comment]
Hello @nipunn1313 !
Any update on that ?
Thanks :)

**Comment by nipunn1313** [comment]
minimum tested requirement is how we've been doing things. I think it's pretty reasonable.

`py_generic_services` - how come that would have to be dropped? I know there are folks using it (using python service impls that are not GRPC). I know dropbox used it for intra-process communication in the desktop client for example.

If it's simply not possible to use, then there's no way around it, but if it is possible to keep, maybe there's another way.

**Comment by nipunn1313** [comment]
```
Replacement: Starting with version 2.3.0, RPC implementations should not try to build on these, but should instead provide code generator plugins which generate code specific to the particular RPC implementation.
```

well that makes sense! I think we can probably put that in the changelog and drop support as well. https://protobuf.dev/news/2024-10-02/#rpc-service-interfaces
Sounds good.
